src/api/events/application-approved-listener.test.js

```diff
@@ -0,0 +1,61 @@
+import * as sqsModule from '@aws-sdk/client-sqs'
+
+// Setup a mock logger object
+const mockLogger = {
+  info: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn()
+}
+
+// Mock the logger module to always return our mockLogger
+jest.mock('~/src/api/common/helpers/logging/logger.js', () => ({
+  createLogger: () => mockLogger
+}))
+
+describe('pollQueue', () => {
+  let sendMock
+  let originalSetInterval
+  let pollQueue
+
+  beforeAll(async () => {
+    // Mock SQSClient.send to return different results based on the command
+    sendMock = jest.fn((command) => {
+      if (command instanceof sqsModule.GetQueueAttributesCommand) {
+        return Promise.resolve({}) // Simulate successful queue verification
+      }
+      if (command instanceof sqsModule.ReceiveMessageCommand) {
+        return Promise.resolve({ Messages: undefined }) // Simulate empty queue
+      }
+      return Promise.resolve({})
+    })
+    sqsModule.SQSClient.prototype.send = sendMock
+
+    // Mock setInterval to call immediately and only once
+    originalSetInterval = global.setInterval
+    global.setInterval = (fn) => {
```

Comment:
I feel like this isn't very "Jesty"... you can use `jest.useFakeTimers()` to use timers you can spy on and then you've got more control over them and manually progress them etc.